### PR TITLE
Analyzer refactor so fortisEvents are inline with cassandra schema

### PR DESCRIPTION
**Pull request by erikschlegel**
This PR encompasses a variety of refactoring changes to ensure the naming conventions are consistent with Cassandra `events` table. Also, had to change the event time data source for all analyzers so it's based off the upstream event time as opposed to the current system time.

### CODE REVIEW

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/dto/FortisEvent.scala
Comment (jcjimenez) [inline src/main/scala/com/microsoft/partnercatalyst/fortis/spark/dto/FortisEvent.scala]: In Cassandra, this is called `event_time` - I'm assuming this discrepancy is ok?

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/locations/Tile.scala
Comment (jcjimenez) [inline src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/locations/Tile.scala:42]: lol

**Review by jcjimenez — APPROVED** [review]
LGTM

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/Pipeline.scala
Comment (kevinhartman) [inline src/main/scala/com/microsoft/partnercatalyst/fortis/spark/Pipeline.scala]: You should move this inside `requiredValuesProvided` as an inner function.

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/BingAnalyzer.scala
Comment (kevinhartman) [inline src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/BingAnalyzer.scala]: `private def`

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/CustomEventAnalyzer.scala
Comment (kevinhartman) [inline src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/CustomEventAnalyzer.scala]: Maybe `custom` would be more useful?

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/ExtendedFortisEvent.scala
Comment (kevinhartman) [inline src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/ExtendedFortisEvent.scala]: We're using 2 space indent as a convention- might be helpful to adjust IDEA's auto formatting to avoid this crazy indent.

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/RedditAnalyzer.scala
Comment (kevinhartman) [inline src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/RedditAnalyzer.scala]: Can `created_utc` be `None`?

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/TadawebAnalyzer.scala
Comment (kevinhartman) [inline src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/TadawebAnalyzer.scala]: Is this the best we can do for Tadaweb?

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/dto/FortisEvent.scala
Comment (kevinhartman) [reply src/main/scala/com/microsoft/partnercatalyst/fortis/spark/dto/FortisEvent.scala]: Yes it's fine. There should be a separate model for DB r/w

**Review by kevinhartman — COMMENTED** [review]
Submitting this batch of comments without explicit LGTM since @erikschlegel mentioned more changes are in progress for this review. Will add second review at that point.

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/RedditAnalyzer.scala
Comment (jcjimenez) [reply src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/RedditAnalyzer.scala]: Yes, but extremely unlikely. I made most fields in `RedditObjectData` optional since different types of objects include different fields (and not using `Optional` was causing the JSON parser to throw). I have a feeling that other stream sources have the potential of exhibiting this problem during JSON parsing.

So, I would like to propose that we create a followup task in order to test by programmatically triggering these  `None`/`null` cases. This way we can ensure that bad/missing network data doesn't cause the entire pipleline to fail if such an exception occurs.

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/sinks/kafka/KafkaSink.scala
Comment (c-w) [inline src/main/scala/com/microsoft/partnercatalyst/fortis/spark/sinks/kafka/KafkaSink.scala]: Woops. Looks like a merge got checked in.

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/sources/streamfactories/FacebookPageStreamFactory.scala
Comment (c-w) [inline src/main/scala/com/microsoft/partnercatalyst/fortis/spark/sources/streamfactories/FacebookPageStreamFactory.scala:22]: Should this just blow up if there are no pageIds? Or would it make sense for us to spin up the Facebook pipeline without any configured pages?

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/ProjectFortis.scala
Comment (c-w) [inline src/main/scala/com/microsoft/partnercatalyst/fortis/spark/ProjectFortis.scala]: Nit: elsewhere ([example](https://github.com/CatalystCode/project-fortis-spark/blob/f609ef8c67f86a62a65ef9208074bdbe0bfb345b/src/main/scala/com/microsoft/partnercatalyst/fortis/spark/ProjectFortis.scala#L133)) we use the pipe character `|` to separate values.

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/BingAnalyzer.scala
Comment (c-w) [inline src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/BingAnalyzer.scala]: Nit: no need to specify the types explicitly. Also: constants should be `PascalCase`.

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/sinks/kafka/KafkaSink.scala
Comment (c-w) [inline src/main/scala/com/microsoft/partnercatalyst/fortis/spark/sinks/kafka/KafkaSink.scala]: Woops. Looks like merge got checked in.

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/locations/Tile.scala
Comment (c-w) [inline src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/locations/Tile.scala]: Why do we need to instantiate the `TileUtils` here? Should we just turn the class into a `object` so that we can access the helpers statically?

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/locations/Tile.scala
Comment (c-w) [inline src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/locations/Tile.scala:40]: I wonder if we should just implement all these tile-utils in Java and add them to [timpark/geotile](https://github.com/timfpark/geotile) so that others can re-use this. Happy to take that if you want.

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/sources/streamfactories/FacebookPageStreamFactory.scala
Comment (erikschlegel) [reply src/main/scala/com/microsoft/partnercatalyst/fortis/spark/sources/streamfactories/FacebookPageStreamFactory.scala:22]: We should raise an exception at the highest severity level to app insights so we're notified. I'll add that.

**Comment by kevinhartman** [comment]
For future PRs, it should be our preference to rebase onto master instead of merging when upstream changes occur (`git pull --rebase origin master`, not `git pull`) in order to keep history clean. It's not interesting to capture merges of concurrent work, and a perfectly sequential history where commits are ordered by corresponding features is easier to read and diff :)

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/FacebookPostAnalyzer.scala
Comment (kevinhartman) [inline src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/FacebookPostAnalyzer.scala:23]: Will this get logged somehow (I suppose it'll print to stderr on the driver)? If not, might make sense to pass to the overload of `logFatalError` that takes `Throwable`.

**Comment by c-w** [comment]
In addition to what @kevinhartman said, you can use `git config --global pull.rebase true` to automatically set up git to always rebase on `git pull`.

File: src/test/scala/com/microsoft/partnercatalyst/fortis/spark/TestFortisEvent.scala
Comment (kevinhartman) [inline src/test/scala/com/microsoft/partnercatalyst/fortis/spark/TestFortisEvent.scala:11]: 2 space indent.

**Review by kevinhartman — APPROVED** [review]
LGTM

**Comment by erikschlegel** [comment]
Re-opening a new pull request off my fork as I agree with @c-w and @kevinhartman that we should only be pushing changes into this repo off forks.

**Pull request closed without merging**
